src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTests.cs

```diff
@@ -72,6 +72,112 @@ public void VerifyMSTestTemplate(DotNetLanguage language)
+    public void VerifyCSharpWPFTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Wpf,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.VB)]
+    public void VerifyVBWPFTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Wpf,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyWebAppTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.WebApp,
+            DotNetSdkActions.Test | DotNetSdkActions.PublishComplex);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyCSharpWinformsTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Winforms,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.VB)]
+    public void VerifyVBWinformsTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Winforms,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyReferenceInConsoleTemplate(DotNetLanguage language)
+    {
+        var referenceTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.ClassLib,
+            DotNetSdkActions.Build);
+        referenceTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Console,
+            DotNetSdkActions.AddClassLibRef | DotNetSdkActions.Build | DotNetSdkActions.Test |
+            DotNetSdkActions.Run | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [MemberData(nameof(GetLanguages))]
+    public void VerifyNET6InConsoleTemplate(DotNetLanguage language)
```

Comment:
I think you should separate the downlevel targeting tests and the latest targeting tests into two sets, and use MemberData to obtain list of downlevel targets, rather than creating a new method for each one. Something like what is done here:

https://stackoverflow.com/questions/22093843/pass-complex-parameters-to-theory

E.g. maybe you might do the following:

```
    [Theory]
    [MemberData(nameof(GetAllLanguagesWithDownlevelTFMs)]
    public void VerifyDownlevelConsoleTemplate(DotNetLanguage language, string framework)
    {
        ...
    }
 
    private static IEnumerable<object[]> GetLanguages() => Enum.GetValues<DotNetLanguage>().Select(lang => new object[] { lang });
    private static string[] AvailableDownLevelTFMs = new string[] { "net6", "net7" };
    private static IEnumerable<object[]> GetAllLanguagesWithDownlevelTFMs() => Enum.GetValues<DotNetLanguage>().SelectMany(lang => AvailableDownLevelTFMs.Select(tfm => new object[] { lang, tfm }));
```

You can use MemberData in various ways to obtain the combination of TFMs and languages you want.

Reply:
For instance, I think it can be made to take parameters (see https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#loading-data-from-a-method-on-the-test-class), so you could add one for "filtered languages".

Reply:
I mostly copied your code here with minor changes for the Downlevel Console test. For the Downlevel WebApp (which only uses 1 language), I opted instead just to use 2 InlineData instead, as that can only use C# for languages. 

```
[Theory]
[InlineData("net6.0")]
[InlineData("net7.0")]
[Trait("Category", "Downlevel")]
public void VerifyDownLevelFrameworksInWebAppTemplate(string frameworks)
```

Let me know if you want something different here.